instrumentation/metric/registry.go

```diff
@@ -85,39 +87,28 @@ func (r *inMemoryRegistry) Get(metricName string) metric {
 			return metric
 		}
 	}
-	panic(errors.Errorf("a metric with the name of %s is not registered", metricName))
+	return nil
 }
 
 func (r *inMemoryRegistry) Remove(m metric) {
 	r.mu.Lock()
 	defer r.mu.Unlock()
-	s := r.mu.metrics
-	for i, existing := range s {
-		if existing.Name() == m.Name() {
-			lastMetric := len(s) - 1
-			s[i] = s[lastMetric]
-			s[lastMetric] = nil // so that it gets garbage-collected, see here https://stackoverflow.com/questions/28832016/re-slicing-and-garbage-collection
-			// We do not need to put s[i] at the end, as it will be discarded anyway
-			r.mu.metrics = s[:lastMetric]
-			return
-		}
+	key := fmt.Sprintf("%p", m)
+	if _, isMetricInMap := r.mu.metrics[key]; isMetricInMap {
+		delete(r.mu.metrics, key)
 	}
-
-	err := errors.Errorf("a metric with name %s cannot be removed as it is not registered", m.Name())
-	panic(err)
 }
 
 func (r *inMemoryRegistry) register(m metric) {
 	r.mu.Lock()
 	defer r.mu.Unlock()
-	for _, existing := range r.mu.metrics {
-		if existing.Name() == m.Name() {
-			err := errors.Errorf("a metric with name %s is already registered", m.Name())
-			panic(err)
-		}
+	key := fmt.Sprintf("%p", m)
```

Comment:
Why not `metric.Key` or something like that? I think it would be more clear

Reply:
i wanted to limit the change. if i add key to the "named metric" i have to change in 3 different specific ctors (gauge, histogram, rate) i just want it limited here. when we do a big change in future we'll change it.

Reply:
maybe add a comment explaining why the pointer is used. I agree it's the right choice here but it's pretty confusing for someone who does not know the story